flink-dist/pom.xml

```diff
@@ -113,8 +113,13 @@ under the License.
 			<artifactId>flink-metrics-jmx</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+        
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-mesos_2.10</artifactId>
+			<version>${project.version}</version>
+		</dependency>
```

Comment:
We only do it for yarn, because Hadoop 2.3 cannot properly support yarn, but we support Hadoop 2.3. Once we drop Hadoop 2.3 support, we can exclude that profile.

I would also build mesos always.